src/codec.rs

```diff
@@ -1108,7 +1110,8 @@ where
 		}
 	}
 
-	decode_vec_chunked(len, |decoded_vec, chunk_len| {
+	input.descend_ref()?;
+	let vec = decode_vec_chunked(len, |decoded_vec, chunk_len| {
```

Comment:
```suggestion
	decode_vec_chunked(len, |decoded_vec, chunk_len| {
```

We are not descending here any further. We just read the straight bytes from the vector.

TBH, the entire "when to descend" is a little bit blurry, but I don't see this case here.

Reply:
Because the idea is just that you want to prevent too much nesting.

Reply:
Ok, I don't know. It seemed like we are still descending in the vec, even though we are reading the bytes. Also seemed strange for example that `Box<u32>` descends, but `Vec<u32>` doesn't. But anyway, I don't know and I don't have strong preferences. Reverted this change.